tests/datavolume_test.go

```diff
@@ -79,6 +78,18 @@ var _ = Describe("DataVolume Integration", func() {
 	}
 
 	Describe("[rfe_id:3188][crit:high][vendor:cnv-qe@redhat.com][level:system] Starting a VirtualMachineInstance with a DataVolume as a volume source", func() {
+		BeforeEach(func() {
+			By("Enable featuregate=HonorWaitForFirstConsumer")
+			patch := `{"spec":{"featureGates":["HonorWaitForFirstConsumer"]}}`
+			_, _, err = tests.RunCommand("kubectl", "patch", "cdiconfigs.cdi.kubevirt.io", "config", "-o=json", "--type=merge", "-p", patch)
```

Comment:
i think you'll want to capture what the cdi config has set before the test, and then only enable/disable based on whether cdi was configured to HonorWaitForFirstConsume. it's possible that setting might be configured as part of the environment setup before the test suite runs.

Reply:
I did try to configure it in `cluster-deploy.sh` script, but unfortunately there are destructive tests (probably this one https://github.com/kubevirt/kubevirt/blob/master/tests/operator_test.go#L1414 this one maybe?) redeploying CDI using api alphav1 and by side-effect resetting new config option. 
That is why after discovering this I have moved enable/disable to before/after test.

Reply:
the enable before/after is fine. 

My point was in the Before(), i'd rather see us capture the original contents on the ConfigMap before we alter it, then in the After() i'd like to see us restore the contents.

This ensures we preserve any featuregates that might have been configured as part of the test setup before the functional tests run. As it is now, the After() function will remove all feature gates.